plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -83,7 +84,22 @@ class NewCreateServiceModalForm extends Component {
     // Hint: When you add something to the state, make sure to update the
     //       shouldComponentUpdate function, since we are trying to reduce
     //       the number of updates as much as possible.
+    // In the Next line we are destructing the config to keep labels as it is and even keep labels with an empty value
+    let { serviceConfig } = ServiceUtil.getServiceJSON(this.props.service);
+    const { labels = {} } = serviceConfig;
 
+    serviceConfig = omit(serviceConfig, ["labels"]);
+
+    let newServiceConfig = Object.assign(
+      {},
+      labels,
+      CreateServiceModalFormUtil.stripEmptyProperties(serviceConfig)
+    );
+    if (Object.keys(labels).length === 0) {
```

Comment:
Why is this if block necessary? What does line 93 produce when `labels = {}`?

Reply:
Line 93 would produce:
```JSON
{labels: {}}
```
The better question is what line 99 produces if labels is `{}`
Because in that case the labels are not present as they are getting stripped.